src/bio/bio_context.c

```diff
@@ -277,10 +277,38 @@ bio_bs_hold(struct bio_blobstore *bbs)
 	return rc;
 }
 
+struct bio_media_error_msg {
+	struct bio_blobstore	*bmem_bs;
+	int			 bmem_rc;
+};
+
+/*
+ * MEDIA ERROR event.
+ * Store BIO I/O error in in-memory device state. Called from device owner
+ * xstream only.
+ */
+static void
+bio_media_error(void *msg_arg)
+{
+	struct bio_media_error_msg	*bmem = msg_arg;
+	struct bio_device_health_state  *hs;
+
+	hs = &bmem->bmem_bs->bb_dev_health.bhm_health_state;
+	hs->bhs_bio_err = bmem->bmem_rc;
+
+	/*
+	 * TODO:
+	 * Mechanism to notify admin of BIO error. Given I/O error, checksum
+	 * error and health state information, admin can determin if device
```

Comment:
sp: determine

Reply:
To simplify the code, I think the faulty criteria check can be done in other function periodically called by owner xstream. This function can do the error logging only.